src/JWT/Builder/JwtHeader.cs

```diff
@@ -8,24 +11,45 @@ namespace JWT.Builder
     public class JwtHeader
     {
         [JsonProperty("typ")]
+#if SYSTEMTEXTJSON

```

Comment:
Should it be one or another (conditionally)?

Reply:
I really don't know .. freedom of choiche is a value ... using netstandard20 I can choose beetween the two types of serializers (like in the asp.net core framework) so why forcing to use System.Text.Json if even microsoft don't force us to do so?

Reply:
Otherwise "replacing" won't be replacing but rather adding one more dependency, what users of Json.Net won't need hence would rightfully question.
Ideally either serializer would need to go to its own satellite package so users would be free to choose between one or another or maybe something third.
My point is that having both attributes and serializers basically means referencing them all and piling their respective attributes on each of this property.
I can easily imagine Unity folks or someone on .NET 3.5 who's unable to use either and who use DataContractJsonSerializer instead coming here and asking to add a third one.
So instead I would switch from a 3rd party to the 1st party serializer (and hope most of the users would be fine with that).

Reply:
yes, I understand ... I leave this choiche to you ... I push another commit with some improvment ...